PutCoin/Models/Transaction.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace PutCoin.Model
+{
+    public class Transaction
+    {
+        public Guid Id { get; set; }
+        public IEnumerable<Guid> OriginTransactionIds { get; set; }
+        public IEnumerable<TransactionDestination> Destinations { get; set; }
+        public string Signature { get; set; }
+        public User User { get; set; }
+
+        public static Transaction GenerateRandomTransaction(BlockChain blockChain, User initiator)
+        {
+            var allTransactions = blockChain.Blocks.SelectMany(x => x.Transactions);
+            var mineTransactions = allTransactions
+                .Where(x => x.Destinations.Select(y => y.Receipent).Contains(initiator))
+                .ToArray();
+
+            //I assume that when I use some transaction I have to use it all
+            var mineNotUsedTransactions = mineTransactions
+                .Where(x => !allTransactions.Any(y => y.OriginTransactionIds.Contains(x.Id) && y.User == initiator))
+                .ToArray();
+
+            var potentialReceipents = allTransactions.SelectMany(x => x.Destinations.Select(y => y.Receipent)).Distinct();
+
+            var originTransaction = mineNotUsedTransactions.Shuffle().FirstOrDefault();
+            if (originTransaction is default(Transaction))
+                return default;
+
+            var valueToSpend = originTransaction.Destinations.Single(x => x.Receipent == initiator).Value;
+
+            var random = new Random();
+            var receipents = potentialReceipents.Shuffle().Take(random.Next(1, 2)).ToArray();
+            List<TransactionDestination> designations = GetDesignationsForNewTransaction(valueToSpend, random, receipents);
+
+            return new Transaction
+            {
+                Destinations = designations,
+                Id = Guid.NewGuid(),
+                OriginTransactionIds = new[] { originTransaction.Id },
+                Signature = initiator.Signature,
+                User = initiator
+            };
+        }
+
+        private static List<TransactionDestination> GetDesignationsForNewTransaction(decimal valueToSpend, Random random, User[] receipents)
```

Comment:
any reason for passing same random? why not just use new ones?

Reply:
I don't get it. If you mean if passing Random object is needed - probably not and there could be just int passed.

Reply:
Done